kapitan/inputs/kadet.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python3
+#
+# Copyright 2018 The Kapitan Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from addict import Dict
+import logging
+from importlib.util import module_from_spec, spec_from_file_location
+import json
+import os
+import sys
+import yaml
+
+from kapitan.inputs.base import InputType, CompiledFile
+from kapitan.resources import inventory as inventory_func
+from kapitan.utils import prune_empty
+
+logger = logging.getLogger(__name__)
+inventory = None
+inventory_global = None
+search_paths = None
+
+
+def module_from_path(path, check_name=None):
+    """
+    loads python module in path
+    set check_name to verify module name against spec name
+    returns tuple with module and spec
+    """
+
+    if not os.path.isdir(path):
+        raise FileNotFoundError("path: {} must be an existing directory".format(path))
+
+    module_name = os.path.basename(os.path.normpath(path))
+    init_path = os.path.join(path, '__init__.py')
+    spec = spec_from_file_location('kadet_component_{}'.format(module_name), init_path)
+    mod = module_from_spec(spec)
+
+    if spec is None:
+        raise ModuleNotFoundError("Could not load module in path {}".format(path))
+    if check_name is not None and check_name != module_name:
+        raise ModuleNotFoundError("Module name {} does not match check_name {}".format(module_name, check_name))
+
+    return mod, spec
+
+
+def load_from_search_paths(module_name):
+    """
+    loads and executes python module with module_name from search paths
+    returns module
+    """
+    for path in search_paths:
+        try:
+            _path = os.path.join(path, module_name)
+            mod, spec = module_from_path(_path, check_name=module_name)
+            spec.loader.exec_module(mod)
+            return mod
+        except ModuleNotFoundError:
```

Comment:
Why not `except (ModuleNotFoundError, FileNotFoundError): pass` here since you handle both anyway in the raise below? Tiny bit cleaner.